test/stores/MemberListStore-test.ts

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { mocked } from "jest-mock";
+import { EventType, IContent, MatrixClient, MatrixEvent, Room } from "matrix-js-sdk/src/matrix";
+
+import SdkConfig from "../../src/SdkConfig";
+import SettingsStore from "../../src/settings/SettingsStore";
+import { MemberListStore } from "../../src/stores/MemberListStore";
+import { stubClient } from "../test-utils";
+import { TestSdkContext } from "../TestSdkContext";
+
+describe("MemberListStore", () => {
+    const alice = "@alice:bar";
+    const bob = "@bob:bar";
+    const charlie = "@charlie:bar";
+    const roomId = "!foo:bar";
+    let store: MemberListStore;
+    let client: MatrixClient;
+    let room: Room;
+
+    beforeEach(() => {
+        const context = new TestSdkContext();
+        client = stubClient();
+        client.baseUrl = "https://invalid.base.url.here";
+        context.client = client;
+        store = new MemberListStore(context);
+        // alice is joined to the room.
+        room = new Room(roomId, client, client.getUserId()!);
+        room.currentState.setStateEvents([
+            new MatrixEvent({
+                type: EventType.RoomCreate,
+                state_key: "",
+                content: {
+                    creator: alice,
+                },
+                sender: alice,
+                room_id: roomId,
+                event_id: "$1",
+            }),
+            new MatrixEvent({
+                type: EventType.RoomMember,
+                state_key: alice,
+                content: {
+                    membership: "join",
+                },
+                sender: alice,
+                room_id: roomId,
+                event_id: "$2",
+            }),
+        ]);
```

Comment:
This doesn't call `Room::recalculate` so `room.getMyMembership()` will not say `join` making this comment borderline false